index.js

```diff
@@ -169,6 +169,23 @@ class Link extends Events {
     })
   }
 
+  startAnnouncing (key, port, opts) {
+    let timeout
+    let stopped = false
+
+    const interval = opts.interval || 2 * 60 * 1000
+    const ann = () => this.announce(key, port, opts, reann)
+    const reann = () => {
+      if (stopped) return
+      timeout = setTimeout(ann, Math.ceil(0.5 * interval + Math.random() * interval))
+    }
+
+    return () => {
+      clearTimeout(timeout)
```

Comment:
can we also make it support stop? https://github.com/bitfinexcom/grenache-nodejs-link/blob/7dd57c82a0e3a03aebed7df2c70d54be50209a2d/index.js#L249

Reply:
good idea, stop should all ongoing announces